File: MExNPrototypes/server/models/Service.js
const mongoose = require('mongoose')

const ServiceSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
    },
    description: {
        type: String,
        required: false
    },
    image: {
        type: String,
        required: false
    },
    tokenCost: {
        type: Number,
        required: true
    },
    registeredBy: {
        type: String,
        required: true
    },
    type: {
        type: string,
        required: true
    }  
})

const Service = mongoose.model("Service", ServiceSchema)
module.exports = Service

File: DOTNETPrototype/Models/Service.cs
namespace Models;

public partial class Service
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public string? Image { get; set; }

    public int TokenCost { get; set; }

    public int? RegisteredBy { get; set; }

    public string Type { get; set; } = null!;

    public virtual User? RegisteredByNavigation { get; set; }

    public virtual ICollection<ServicePurchase> ServicePurchases { get; set; } = new List<ServicePurchase>();
}
